fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -0,0 +1,359 @@
+            ImmutableMap.of(IndexOptions.TEXT_TOKENIZER_NAME_OPTION, AllSuffixesTextTokenizer.NAME));
+
+    private static final String DYLAN = "You're an idiot, babe\n" +
+                                        "It's a wonder that you still know how to breathe";
+
+    private static final String WAYLON = "There's always one more way to do things and that's your way, and you have a right to try it at least once.";
+
+    protected void openRecordStore(FDBRecordContext context, FDBRecordStoreTestBase.RecordMetaDataHook hook) {
+        RecordMetaDataBuilder metaDataBuilder = RecordMetaData.newBuilder().setRecords(TestRecordsTextProto.getDescriptor());
+        metaDataBuilder.getRecordType(COMPLEX_DOC).setPrimaryKey(concatenateFields("group", "doc_id"));
+        hook.apply(metaDataBuilder);
+        recordStore = getStoreBuilder(context, metaDataBuilder.getRecordMetaData())
+                .setSerializer(TextIndexTestUtils.COMPRESSING_SERIALIZER)
+                .uncheckedOpen();
+        setupPlanner(null);
+    }
+
+    @Nonnull
+    protected FDBRecordStore.Builder getStoreBuilder(@Nonnull FDBRecordContext context, @Nonnull RecordMetaData metaData) {
+        return FDBRecordStore.newBuilder()
+                .setFormatVersion(FDBRecordStore.MAX_SUPPORTED_FORMAT_VERSION) // set to max to test newest features (unsafe for real deployments)
+                .setKeySpacePath(path)
+                .setContext(context)
+                .setMetaDataProvider(metaData);
+    }
+
+    private TestRecordsTextProto.SimpleDocument createSimpleDocument(long docId, String text, int group) {
+        return TestRecordsTextProto.SimpleDocument.newBuilder()
+                .setDocId(docId)
+                .setText(text)
+                .setGroup(group)
+                .build();
+    }
+
+    private TestRecordsTextProto.ComplexDocument createComplexDocument(long docId, String text, String text2, int group) {
+        return TestRecordsTextProto.ComplexDocument.newBuilder()
+                .setDocId(docId)
+                .setText(text)
+                .setText2(text2)
+                .setGroup(group)
+                .build();
+    }
+
+
+    @Test
+    public void simpleInsertAndSearch() {
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context, metaDataBuilder -> {
+                metaDataBuilder.removeIndex(TextIndexTestUtils.SIMPLE_DEFAULT_NAME);
+                metaDataBuilder.addIndex(SIMPLE_DOC, SIMPLE_TEXT_SUFFIXES);
+            });
+            recordStore.saveRecord(createSimpleDocument(1623L, DYLAN, 2));
+            recordStore.saveRecord(createSimpleDocument(1547L, WAYLON, 1));
+            RecordCursor<IndexEntry> indexEntries = recordStore.scanIndex(SIMPLE_TEXT_SUFFIXES, BY_LUCENE_QUERY, TupleRange.allOf(Tuple.from("idiot")), null, ScanProperties.FORWARD_SCAN);
```

Comment:
It looks like `FORWARD_SCAN` is another thing ignored by the actual index maintainer / scan logic.